### DISCUSSION TITLE
tutorials/lift-off-part2/01-journey-of-a-graphql-query

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#39] [course feedback]
# tutorials/lift-off-part2/01-journey-of-a-graphql-query

https://www.apollographql.com/tutorials/lift-off-part2/01-journey-of-a-graphql-query

**Comment by flipsflops** [comment]
Love this so much ! Thank you for putting the time and effort for all of this.

**Comment by CMH7** [comment]
I love the way the visuals are done! <3 It is much easier to understand with what is going on.

**Comment by vasily-mishanin** [comment]
The explanation is brilliant!

**Comment by OliverFarren** [comment]
The animated visual on this tutorial is amazing.

**Comment by krichard-aurorasolar** [comment]
This one is my favourite one so far.

**Comment by IstvanCsVarga** [comment]
This is the best training material I've seen in a while! Amazing job!

**Comment by marksikaundi** [comment]
I think am here to learn graphql and make it

**Comment by LahiruRajapaksha** [comment]
This is awesome

**Comment by Femtech-web** [comment]
The best tutorial I have ever taken

**Comment by belloshehu** [comment]
Understanding the client-server request and response round-trip in GraphQL is interesting! Does it mean that all field must have resolvers that assign value to their fields?

**Reply by lizhennessy** [reply]
Hi @belloshehu ! It depends on the tool that you're using - Apollo Server is able to define [_default resolvers_](https://www.apollographql.com/docs/apollo-server/data/resolvers/#default-resolvers) which means we don't have to define a function for every field. Check out the article I linked for a helpful flowchart on when default resolvers are created!

**Reply by belloshehu** [reply]
It is really cool having to learn that Apollo lifts so many things for us. Thanks for the clarification

**Comment by repiemm** [comment]
The visual material has helped me a lot to understand the concepts, great work!

**Comment by aygulpardane** [comment]
Hello - could you kindly share resources on connecting GraphQL to a database, or specifically Postgres? Thank you!

**Reply by mabuyo** [reply]
Hey there @aygulpardane - I searched through our Apollo resources but haven't found anything. You can try asking in our Discord group -> https://discord.gg/graphos

If anyone sees this comment here as well, please feel free to share any resources that have helped you ❤️

**Comment by NishantCoder108** [comment]
# Journey of a GraphQL query ! Start 🚀🚀🚀

**Comment by sylvio-cavalcanti** [comment]
This course is really something else, great work so far, awesome animation and explanation!

**Comment by gustvoassuncao** [comment]
Awesome as always, I've been learning so much, thanks!

**Comment by jimmcclarty** [comment]
Seems the images in the sample project are no longer hosted at cloudinary
<img src="https://res.cloudinary.com/dety84pbu/image/upload/v1598465568/nebula_cat_djkt9r.jpg" alt="Astro Kitty, Space Explorer" class="css-jj20hb">
https://res.cloudinary.com/dety84pbu/image/upload/v1598465568/nebula_cat_djkt9r.jpg

**Reply by oayras** [reply]
here same

**Reply by onlyroz** [reply]
Yes - same for me - I copied the mock from part 1 and that worked

**Comment by MohammadEzz** [comment]
Awesome tutorial

**Comment by sichi46** [comment]
Masterpiece!
